connectionmanager/connectionmanager.go

```diff
@@ -132,7 +132,13 @@ func connect(
 		mutex:      &sync.Mutex{},
 	}
 
-	log.Tracef("connectionmanager connect returning")
+	// TODO: to be tested, did this really fix the deadlock?
+	select {
+	case <-cancelReconnectChan[proto]:
+		log.Trace("connectionmanager drained cancelReconnectChan")
```

Comment:
Question:
It's not necessarily drained though, just one element taken out?

Reply:
This I am not very sure about. Only one element is taken out but is only one place where elements are put in, so it is "drained" because is unbuffered.

I think that having moved the sender (line 222) to a goroutine is enough and we can remove this part here. If it exists with a pending element in the channel here it will only be a problem next time the reconnection is attempted to be cancelled, as it will timeout there.

But it should be harmful anyway, so I decided to leave with trace logging.